tests/backup/backup_test.go

```diff
@@ -303,6 +309,232 @@ func TeardownForTestcase(contexts []*scheduler.Context, providers []string, Clou
+		})
+
+		Step("Taking backup of applications", func() {
+			var sem = make(chan struct{}, 10)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range bkpNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					time.Sleep(1 * time.Second)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+							labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Sharing backup with groups", func() {
+			backupsToBeSharedWithEachGroup := 3
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for i, backupName := range backupNames {
+				groupIndex := i / backupsToBeSharedWithEachGroup
+				switch groupIndex {
+				case 0:
+					ShareBackup(backupName, []string{groups[groupIndex]}, nil, ViewOnlyAccess, ctx)
+				case 1:
+					ShareBackup(backupName, []string{groups[groupIndex]}, nil, RestoreAccess, ctx)
+				case 2:
+					ShareBackup(backupName, []string{groups[groupIndex]}, nil, FullAccess, ctx)
+				default:
+					ShareBackup(backupName, []string{groups[0]}, nil, ViewOnlyAccess, ctx)
+				}
+			}
+		})
+
+		Step("Share Backup with Full access to a user of View Only access group and Validate", func() {
+			//TODO: Add validation logic
+		})
+
+		Step("Share Backup with View Only access to a user of Full access group and Validate", func() {
+			//TODO: Add validation logic
+		})
+
+		Step("Share Backup with Restore access to a user of View Only access group and Validate", func() {
+			//TODO: Add validation logic
+		})
+
+		Step("Validate Restore access for a user of Restore group", func() {
+			//TODO: Add validation logic
+		})
+
+		Step("Validate that user with View Only and Restore access cannot delete the backup", func() {
+			//TODO: Add validation logic
+		})
+	})
+	JustAfterEach(func() {
+		var wg sync.WaitGroup
+		defer EndTorpedoTest()
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		backupDriver := Inst().Backup
+		for _, backupName := range backupNames {
+			backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+			log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
+			DeleteBackup(backupName, backupUID, orgID, ctx)
+		}
+		DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID)
+		time.Sleep(time.Minute * 3)
```

Comment:
same here

Reply:
Same comment as above. Want to address in next PR.